src/plugins/intel_gpu/src/kernel_selector/kernels/dynamic_quantize/dynamic_quantize_kernel_opt.cpp

```diff
@@ -6,7 +6,22 @@
 #include "kernel_selector_utils.h"
 #include <string>
 
+
+static constexpr size_t simd = 16;
+
 namespace kernel_selector {
+static size_t get_match_vector_size(const dynamic_quantize_params& params) {
+    auto block_sizes = { 8, 4, 2 };
+
+    for (auto block_size : block_sizes) {
+        if (((params.inputs[0].X().v * params.inputs[0].Y().v) / simd) % block_size == 0) {
+            return block_size;
+        }
+    }
+
+    return 1;
```

Comment:
does opt kernel works with `vec_size=1`?

Reply:
Now validate functions returns false if vec_size is 1.